### PR TITLE
Added deploy scripts that upgrade/reboot the nodes too

### PR DESCRIPTION
**Pull request by andywis**

### CODE REVIEW

**Review by stevemul — COMMENTED** [review]
I think stage 3 and stage 4 need to be swapped around. It'd be better to patch the underlying hosts and reboot them prior to deploying the OpenShift containers on top.

Other than that it looks good to me - can we test it through the pipeline? We'll need to think about how we can add a test to verify there aren't any outstanding patches prior to deployment.

**Comment by andywis** [comment]
refactor: we need "scaleup.sh" and "scaleup.yml" (a playbook) and probably some Python scripts in tools/

**Comment by stevemul** [comment]
As discussed, the reboot needs to happen before the deployment, should've covered it in my previous comment, apologies. So that means swapping stages 4 and 5 around now.

**Comment by andywis** [comment]
Stages 4 and 5 are now swapped round

**Comment by stevemul** [comment]
Please retest.

**Comment by stevemul** [comment]
@andywis Did you manage to test this one? It'd be good to get this code merged and pull request closed off.

**Comment by andywis** [comment]
This has become a mess since I rebased.

**Comment by andywis** [comment]
The important files to look at are:
* deploy-with-upgrade.sh (which will be merged with deploy-openshift.sh eventually)
* upgrade_and_reboot_servers.yml
* .gitignore

**Comment by andywis** [comment]
Attempting to tidy the pull request as per https://stackoverflow.com/a/26986320

**Comment by andywis** [comment]
Pull request is now much tidier now.

**Comment by stevemul** [comment]
Where are we with this @andywis? Is it tested and ready for merge?

**Comment by andywis** [comment]
tested and working. Ready to merge. Relies on  PR#38 in the Heat repo

File: sanity_tests.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+#
+GROUP_VARS_FILE=group_vars/all.yml
+
+function do_we_have_ip_addresses {
+   for node in haproxy-0 haproxy-1 master-0 worker-0; do
+       grep ${node} ${GROUP_VARS_FILE} | grep -E "([0-9]{1,3}\.){3}[0-9]{1,3}" > /dev/null
+       if [[ $? -ne 0 ]]; then
+           echo "ERROR: ${node} does not have a valid IP address."
+           echo "(This can be caused if you have another stack in your tenancy, stealing your IP Addresses.)"
```
Comment (stevemul) [inline sanity_tests.sh]: This is a reference to a specific heat deployment issue - bear in mind this is supposed to be separate from the heat code, so whilst its valid to check the correct config is in place, if this is deployed on another cloud provider or in vCenter for a non cloud deployment then this comment isn't relevant and could be considered confusing. It'd be better to stick to what should be there and why its needed and let the person deploying troubleshoot why.

**Review by stevemul — COMMENTED** [review]
Minor comment only

**Review by stevemul — APPROVED** [review]
lgtm

**Pull request merged**